x509/testdata/onionSANBadServDescAlgorithm.pem

```diff
@@ -0,0 +1,12 @@
+-----BEGIN CERTIFICATE-----
```

Comment:
I'm not going to make you provide a programmatic method for generating these test certificates, but if you do have any scripting or even a link to some certs on Censys, just drop it in this directory.

Reply:
I'm afraid I don't :-( I made these certs by hand tweaking a cert template in a one-off Go program. I didn't save intermediate versions of the program as I went along, just the rendered PEM encoded certificates. In retrospect that was pretty hackish.

I think rather than provide scripting/links to test certs it would be cleaner overall to re-implement the tests as in-memory `pkix.Extension` instances handed directly to `parseTorServiceDescriptorSyntax` instead of PEM encoded certificates given to `ParseCertificate`. I mostly did the latter because that's how I wrote the `zlint` tests where it made more sense to operate at the certificate level. If I do the former I think the tests will be self-documenting and won't rely on opaque test data.

Would that work for you?

Reply:
I implemented the above in 9458ee5 and deleted the `.pem` files. The testcase code I wrote is now pretty much identical to the relevant parts of the throw away go programs I wrote to generate the `.pem`. I think this is the better approach, thanks for the nudge! :+1: